src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -824,6 +824,11 @@ void MarkDynamicDependency (DynamicDependency dynamicDependency, IMemberDefiniti
 					_context.LogWarning ($"Unresolved type '{typeName}' in DynamicDependencyAttribute", 2036, context);
 					return;
 				}
+
+				var module = assembly.MainModule;
+				if (module.GetMatchingExportedType (type, out var exportedType)) {
```

Comment:
Why do you need to do this when you have correct data in dynamicDependency.AssemblyName ?